### PR TITLE
[SecurityBundle] make remember-me cookies auto-secure + inherit their default config from framework.session.cookie_*

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28338
| License       | MIT
| Doc PR        | -

Let's make it easier to have a good default security level, now for the remember-me cookie.

### CODE REVIEW

**Review by leofeyer — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php [hunk 1]
```diff
@@ -54,6 +56,34 @@ public function __construct()
         }
     }
 
+    public function prepend(ContainerBuilder $container)
+    {
+        $rememberMeSecureDefault = false;
+        $rememberMeSameSiteDefault = null;
+
+        foreach ($container->getExtensions() as $name => $extension) {
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: `if (isset($container->getExtensions()['framework'])) {` instead?

**Review by fabpot — CHANGES_REQUESTED** [review]
Missing some notes in the changelog files.

**Comment by stof** [comment]
As you add support in BrowserKit, shouldn't you also not send the cookie in the request if it is a samesite=strict one and we don't have a request to the same site as the last one in the history ? Otherwise, we don't actually support them.

**Comment by nicolas-grekas** [comment]
Comments addressed.

Status: needs review

File: src/Symfony/Bundle/SecurityBundle/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ CHANGELOG
    or `Symfony\Component\Security\Core\Authentication\Token\RememberMeToken`.
  * Added `Symfony\Bundle\SecurityBundle\DependencyInjection\Compiler\AddExpressionLanguageProvidersPass`
  * Added `json_login_ldap` authentication provider to use LDAP authentication with a REST API.
+ * Made remember-me cookies inherit their default config from `framework.session.cookie_*`
+   and added an "auto" mode to their "secure" config option to make them secure on HTTPS automatically.
+
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]: extra blank line

Comment (nicolas-grekas) [reply src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]: fixed thanks :)

**Review by leofeyer — APPROVED** [review]
Except the two unrelated changes (`src/Symfony/Component/DomCrawler/CHANGELOG.md` and `src/Symfony/Component/Finder/CHANGELOG.md`), this look good to me.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**